### PR TITLE
Webumenia 819 store and display vp start date and vp end date fo

### PR DESCRIPTION
**Pull request by FrantisekMichalSebestyen**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (link to Jira or GitHub issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [ ] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

Snímka obrazovky 2022-09-14 o 16 43 33

### CODE REVIEW

**Comment by DigitalArtsAFAD** [comment]
what's next? @mcudrnak @FrantisekMichalSebestyen ?

**Review by eronisko — CHANGES_REQUESTED** [review]
The fact that we've created 3 static functions (but we're only using one of them) hints to me that this perhaps doesn't belong to Authority and it's more presentation-related.

I'd suggest moving this into a [Blade component](https://laravel.com/docs/9.x/blade#components). (You can also [define methods there](https://laravel.com/docs/9.x/blade#component-methods))

**Comment by eronisko** [comment]
Please also check if your Prettier is on (it doesn't seem to be)

**Comment by FrantisekMichalSebestyen** [comment]
> Please also check if your Prettier is on (it doesn't seem to be)

You're right, I'm just confused what it the policy for formatting now? 
Should I at the end make commit formatting all affected files ?

File: app/Authority.php [hunk 1]
```diff
@@ -373,6 +382,27 @@ public static function formatMultiAttribute($atttribute, $index = 0)
...
+        if(in_array($activity, $ommitedActivities))
+        {
+            $activity = null;
+        }
+        return $activity;
+    }
+
+    public static function formatEvent($event)
+    {
+        $activity = self::formatActivity(self::formatMultiAttribute($event->event));
```
[the remark this answers is not in the document]
Comment (FrantisekMichalSebestyen) [reply app/Authority.php:397]: I felt like inlining the function into blade component was a bit overwhelming to read, but I will move it.

**Comment by eronisko** [comment]
> > Please also check if your Prettier is on (it doesn't seem to be)
> 
> You're right, I'm just confused what it the policy for formatting now? 

I wouldn't call it a "policy" but I usually 

* format everything (prettier) if I've made big changes
* not format if I've only changed a few lines in an unformatted file

> Should I at the end make commit formatting all affected files ?

That's a good idea if you can put those changes in a 'formatting-only' commit. Preferably at the start, though.

The Right Solution™ I guess is to just format everything as we discussed on Slack.

File: resources/views/components/authority-event.blade.php [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@props(['event', 'count' => 0, 'i' => 0])
+<strong><a
+        href="{!! route('frontend.author.index', ['place' => $event->place]) !!}">{!! $event->place !!}</a></strong>{{ $formatEvent($event) }}{{ $i + 1 < $count ? ', ' : '' }}
```
Comment (eronisko) [inline resources/views/components/authority-event.blade.php:3]: Let's move `{{ $i + 1 < $count ? ', ' : '' }}` out of here, there's no need to for this component to be concerned with its outside context. (Don't forget to remove attributes from `@props` too)

**Review by eronisko — CHANGES_REQUESTED** [review]
Almost there

File: app/View/Components/AuthorityEvent.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        return $start_date . ' - ' . $end_date;
+    }
+
+    public function formatEvent($event)
+    {
+        $activityTranslation = Authority::formatMultiAttribute($event->event);
+        $filteredActivity = in_array($activityTranslation, ['pôsobenie', 'activity'])
+            ? null
+            : $activityTranslation;
```
[the remark this answers is not in the document]
Comment (FrantisekMichalSebestyen) [reply app/View/Components/AuthorityEvent.php:32]: They are not equal, tho. The recommended solution does not work for authorities, that only have one translated "role".

[the remark this answers is not in the document]
Comment (eronisko) [reply app/View/Components/AuthorityEvent.php:32]: You're right, thanks

**Comment by FrantisekMichalSebestyen** [comment]
@eronisko ready for review

**Pull request merged**